packages/aws-cdk-lib/aws-sns-subscriptions/lib/sqs.ts

```diff
@@ -38,6 +38,18 @@ export class SqsSubscription implements sns.ITopicSubscription {
     }
     const snsServicePrincipal = new iam.ServicePrincipal('sns.amazonaws.com');
 
+    // if the queue is encrypted by AWS managed KMS key (alias/aws/sqs),
+    // throw error message
+    if ((this.queue.node.defaultChild as sqs.CfnQueue).kmsMasterKeyId == 'alias/aws/sqs') {
```

Comment:
We also need to handle the case where the queue is imported `IQueue` in which
case there will not be a default child.

I think we may want to add a new attribute to the `IQueue`, something like
`encryptionType` that holds the type of encryption.

Reply:
Hi @corymhall, thank you for your review! I understand your concern.

If users import the queue via `fromQueueArn` method, CDK cannot determine whether the queue is encrypted by AWS managed KMS key or not.
https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-sqs/lib/queue.ts#L268-L287

So in my understanding, we can't validate it completely in that case. I think one possible solution is skipping the validation if `encryptionMasterKey` is undefined to prevent error due to missing default child. Does it make sense?

Reply:
I believe users can provide the key in `fromQueueAttributes` so if the `encryptionType` is undefined then we don't do the validation.

Reply:
Yeah I think so.